jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java

```diff
@@ -16,22 +16,36 @@
 package ml.dmlc.xgboost4j.java;
 
 import java.io.*;
+import java.lang.reflect.Constructor;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-import com.esotericsoftware.kryo.Kryo;
-import com.esotericsoftware.kryo.KryoSerializable;
-import com.esotericsoftware.kryo.io.Input;
-import com.esotericsoftware.kryo.io.Output;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
 /**
  * Booster for xgboost, this is a model API that support interactive build of a XGBoost Model
  */
-public class Booster implements Serializable, KryoSerializable {
+public class Booster implements Serializable {
+
   private static final Log logger = LogFactory.getLog(Booster.class);
+
+  private static final boolean USE_KRYO_BOOSTER;
+  private static final Class<?> KRYO_BOOSTER_CLASS;
+  static {
+    Class<?> kryoBoosterClass;
+    try {
+      String kryoBoosterClassName = Booster.class.getPackage().getName() + ".KryoBooster";
+      kryoBoosterClass = Class.forName(kryoBoosterClassName);
```

Comment:
Quick questions:
  - using caller's CL makes sense here for me, however, is there any situation that we need to specify explicit CL?
  - do you want to initialize class by default? Are there any side-effect in initializing  Kryo API?

Reply:
Regarding explicit CL - I don't know, I don't see the use case right now.

Static initialization seems fine to me because it basically mimics how was the class loading handled before. It can have side effects because to load KryoBooster you need KryoSerializable and Input/Output classes - initializing these can have side effects but I think it is a good side effect - it preserves the former behavior.